modyn/selector/internal/selector_strategies/abstract_selection_strategy.py

```diff
@@ -46,6 +46,20 @@ def __init__(
         self.training_set_size_limit: int = config["limit"]
         self.has_limit = self.training_set_size_limit > 0
         self.reset_after_trigger: bool = config["reset_after_trigger"]
+        if "tail_triggers" in config:
+            self.tail_triggers = config["tail_triggers"]
+            assert not (
+                self.reset_after_trigger and self.tail_triggers > 0
+            ), "Reset after trigger is equivalent to setting tail triggers to 0."
+            assert not (
+                not self.reset_after_trigger and self.tail_triggers == 0
+            ), "Reset after trigger is equivalent to setting tail triggers to 0."
+        else:
+            if self.reset_after_trigger:
+                self.tail_triggers = 0  # consider only the current trigger
+            else:
+                self.tail_triggers = -1  # consider every datapoint
+
```

Comment:
Please add in all existing strategies checks that self.tail_triggers == -1 or not given at all, so we don't ignore it silently.

Reply:
I implemented it on all methods except freshness sampling, on which I put the check